R/write.R

```diff
@@ -50,13 +50,45 @@
 #'
 #' @export
 #' @importFrom basilisk basiliskRun
+#' @importFrom Matrix sparseMatrix
 writeH5AD <- function(sce, file, X_name = NULL, skip_assays = FALSE) {
+    # Loop over and replace DelayedArrays.
+    ass_list <- assays(sce)
+    is_da <- logical(length(ass_list))
+    for (a in seq_along(ass_list)) {
+        if (is(ass_list[[a]], "DelayedMatrix")) {
+            is_da[a] <- TRUE
+            assay(sce, a, withDimnames=FALSE) <- sparseMatrix(i=integer(0), 
+                j=integer(0), x=numeric(0), dims=dim(sce))
+        }
+    }
+
     file <- path.expand(file)
     basiliskRun(
         env = anndata_env,
         fun = .H5ADwriter,
         sce = sce, file = file, X_name = X_name, skip_assays = skip_assays
     )
+
+    # Going back out and replacing each of them.
+    if (any(is_da)) {
+        paths <- file.path("layers", assayNames(sce)[is_da])
+        if (is_da[1]) {
+            paths[1] <- "X"
+        }
```

Comment:
Is this redundant with the check within the loop? Just wondering if we could cut one of them?

Reply:
Yeah, probably. Must have started writing it outside the loop and realized I needed to move it inside and forgot to delete the former. I'd just move the `file.path` call inside the loop as well, no point splitting the logic.